src/_guides/language/async-await.md

```diff
@@ -0,0 +1,344 @@
+
+
+## Practice Handling Errors with async and awaitImplement code to accomplish the following:
+
+
+
+* Make `changeUsername` call the provided asynchronous function `getNewUsername`, and return its result
+* `changeUsername` must catch and return any errors thrown by getNewUsername, prefixing the returned error with the following string: `Error:` 
+* Example return value from `changeUsername` when `getNewUsername` throws an error: `Error: Username must contain only alphanumeric values`
+
+
+## Putting It All Together
+
+
+
+Now it's time to sum up what you've learned in one final exercise. In the following snippet, write the following:
+
+* Part1: Write a function `addHello.` `addHello` should take a single string argument, and return that argument surrounded by the text ‘Hello <string>!' 
+* Part2: Write a function `greetUser` that takes no arguments
+* To obtain the username, `greetUser` should call the provided async function `getUsername`, which waits for 1 second before returning a string
+* `greetUser` should create a greeting for the user by calling `addHello,` passing it the username, and returning the result. For example, if the username is ‘Jenny' `greetUser` should create and return the greeting "Hello Jenny!"
+* Part3: Write a function `sayGoodbye` that takes no arguments
+* Within its function body, `sayGoodbye` should call the provided async function `logoutUser` which takes no arguments.
+* `sayGoodbye` should catch any errors thrown by `logoutUser`
+* If `logoutUser` succeeds, `sayGoodbye` should return the string "<result> Thanks! See you next time!" where <result> is the String value returned by calling `logoutUser`
+
+<aside class="special">
+
+**Note**: All Dart code runs in the context of an  [isolate](https://api.dartlang.org/stable/2.2.0/dart-isolate/dart-isolate-library.html) that owns all of the memory that the Dart code uses. While Dart code is executing, no other code in the same isolate can run.
+
+If you want multiple parts of Dart code to run concurrently, you can run them in separate isolates. (Web apps use  *workers*  instead of isolates.) Multiple isolates run at the same time, usually each on its own CPU core. Isolates don't share memory, and the only way they can interact is by sending messages to each other. 
+</aside>
+
+Find Out More
+
+*  [Futures](https://www.dartlang.org/tutorials/language/futures)
+*  [The Event Loop in Dart](https://webdev.dartlang.org/articles/performance/event-loop)
+*  [Isolates](https://api.dartlang.org/stable/2.2.0/dart-isolate/dart-isolate-library.html)
+* Read about  [Dart's type system](https://dart.dev/guides/language/sound-dart) for more examples of type signatures with futures. You should  [prefer signatures in function type annotations](https://dart.dev/guides/language/effective-dart/design#prefer-signatures-in-function-type-annotations) wherever possible!
```

Comment:
Use relative links, not absolute links, when possible. In this case:

[Dart's type system](/guides/lanuage/sound-dart) for ....

I know that you link to Effective Dart for "preferring signatures", but can you summarize, in one sentence, why? I clicked over to the link and it's not really clear why. (And if you can't, maybe this advice should be deleted here.)

Reply:
I'm not sure that "relative" is the right term here, but yep, never use "dart.dev" for anything in dart.dev. If you think the page might move, you could use a macro instead (`{{site.dartlang}}` or  something like that.)